source/common/http/header_utility.h

```diff
@@ -290,6 +290,19 @@ class HeaderUtility {
    */
   static std::string addEncodingToAcceptEncoding(absl::string_view accept_encoding_header,
                                                  absl::string_view encoding);
+
+  /**
+   * Return `true` if the request is a standard HTTP CONNECT.
+   * HTTP/1 RFC: https://datatracker.ietf.org/doc/html/rfc9110#section-9.3.6
+   * HTTP/2 RFC: https://datatracker.ietf.org/doc/html/rfc9113#section-8.5
+   */
+  static bool isStandardConnectRequest(const Http::RequestHeaderMap& headers);
+
+  /**
+   * Return `true` if the request is an extended HTTP/2 CONNECT.
+   * according to https://datatracker.ietf.org/doc/html/rfc8441#section-4
+   */
+  static bool isExtendedH2ConnectRequest(const Http::RequestHeaderMap& headers);
```

Comment:
now that I look at this, the problem of this versus a state boolean, is it'll not consistently work because of transformations.
not sure what to do about it at this point :-/

Reply:
Let's discuss it further. I'll make a note for myself to address this.